### PR TITLE
EVG-18517: Add Execution label Tasks and Task Duration Table

### PR DESCRIPTION
**Pull request by SupaJoon**
[EVG-18517](https://jira.mongodb.org/browse/EVG-18517)
[Figma Design](https://www.figma.com/file/siN3h8eTcYmkYU3mxBcENt/Evergreen-Version-1-(Updated-Verion)?node-id=1406%3A12619&t=9TuAkgwD8QjxAzf5-0)
### Description
`showExecutionLabel` on `<TasksTable />` determines whether the label should be rendered and `<ExecutionTasksTable />` will calculate the prop. I included a story for Task Duration since it didn't exist before.

### CODE REVIEW

File: src/components/Table/TaskLink.tsx [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 export const TaskLink: React.VFC<TaskLinkProps> = ({
+  execution,
+  onClick = () => {},
   taskId,
   taskName,
-  onClick = () => {},
 }) => (
   <StyledRouterLink onClick={() => onClick(taskId)} to={getTaskRoute(taskId)}>
     <WordBreak>{taskName}</WordBreak>
+    <Body>Execution {executionAsDisplay(execution)}</Body>
```
Comment (sophstad) [inline src/components/Table/TaskLink.tsx:20]: We only need to show this label on the execution tasks table (not a patch's tasks table) since the project focuses on restarting failed execution tasks thereby making those runs out of sync. I think the designs show the label on the wrong table so I just checked with Annie and confirmed this is the case.

I also think we could maybe only show the label if there is more than one execution for any execution task, but maybe that's something we should confirm in design review?

**Review by sophstad — CHANGES_REQUESTED** [review]

File: src/components/Table/TaskLink.tsx [hunk 2]
```diff
@@ -1,17 +1,24 @@
+import { Body } from "@leafygreen-ui/typography";
 import { StyledRouterLink, WordBreak } from "components/styles";
 import { getTaskRoute } from "constants/routes";
+import { executionAsDisplay } from "pages/task/util/execution";
```
Comment (khelif96) [inline src/components/Table/TaskLink.tsx:4]: Should we move this to a more general location so we aren't importing from page directories?

perhaps "utils/task"

File: src/components/Table/TasksTable.stories.tsx [hunk 3]
```diff
@@ -39,55 +41,11 @@ const tasks = [
...
-        blocked: false,
-        baseTask: {
-          status: "system-failed",
-        },
-      },
-    ],
-  },
-];
 export const BaseTaskTable = () => <TasksTable tasks={tasks} />;
-export const ExecutionTasksTable = () => <TasksTable tasks={nestedTasks} />;
```
Comment (khelif96) [inline src/components/Table/TasksTable.stories.tsx:90]: Why did you delete this story I think there's value in having both stories since we no longer can test collapsing and uncollapsing.

File: src/components/Table/TasksTable.tsx [hunk 4]
```diff
@@ -22,45 +22,46 @@ import { TableOnChange } from "types/task";
...
+  displayName: string;
   executionTasksFull?: TaskTableInfo[];
+  id: string;
   projectIdentifier?: string;
+  status: string;
 };
 
 interface TasksTableProps {
   baseStatusSelectorProps?: TreeSelectProps;
+  loading?: boolean;
```
Comment (khelif96) [inline src/components/Table/TasksTable.tsx:39]: Thanks for sorting these!

File: src/pages/task/taskTabs/ExecutionTasksTable.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  },
+];
+export const SingleExecution = () => (
+  <ExecutionTasksTable executionTasksFull={singleExecution} execution={5} />
+);
+export const MultipleExecutions = () => (
+  <ExecutionTasksTable executionTasksFull={multipleExecutions} execution={14} />
+);
+export default {
+  title: "Components/Execution Tasks Table",
```
Comment (khelif96) [inline src/pages/task/taskTabs/ExecutionTasksTable.stories.tsx:51]:
```suggestion
  title: "Pages/Task/Table/Execution Tasks Table",
```

File: src/pages/version/taskDuration/TaskDurationTable.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    },
+  ],
+  loading: false,
+};
+
+export const Default = () => (
+  <TaskDurationTable tasks={props.tasks} loading={false} />
+);
+export default {
+  title: "Components/Task Duration Table",
```
Comment (khelif96) [inline src/pages/version/taskDuration/TaskDurationTable.stories.tsx:113]:
```suggestion
  title: "Pages/Task/Table/Task Duration Table",
```

**Review by khelif96 — CHANGES_REQUESTED** [review]

**Review by sophstad — APPROVED** [review]

File: src/components/Table/TasksTable.tsx [hunk 7]
```diff
@@ -132,8 +137,13 @@ const getColumnDefs = ({
...
     width: "40%",
     className: "cy-task-table-col-NAME",
-    render: (name: string, { id }: Task): JSX.Element => (
-      <TaskLink onClick={onClickTaskLink} taskName={name} taskId={id} />
+    render: (name: string, { id, execution }: Task): JSX.Element => (
+      <TaskLink
+        onClick={onClickTaskLink}
+        taskName={name}
+        taskId={id}
+        execution={showTaskExecutionLabel ? execution : null}
```
[the remark this answers is not in the document]
Comment (khelif96) [reply src/components/Table/TasksTable.tsx:145]: I agree it will make it easier to do this as well. https://jira.mongodb.org/browse/EVG-18647

**Review by khelif96 — APPROVED** [review]
LGTM my only suggestion is to make the change Sophie suggested above.
